### PR TITLE
Added prow-workloads overlay and deployment job

### PR DESCRIPTION
**Pull request by fgimenez**
Modifies prow-deploy to be able to deploy the new cluster components. It also adds a postsubmit to run the actual deployment on tag push.

It also includes these fixes that affect the deployment on ibm-prow:
* Fixed gcs secret file name
* Fixed project-infra root path (fgimenez -> kubevirt)
* Added kubeconfig path to Kustomize command
* Updated to use GIT_ASKPASS instead of GITHUB_TOKEN

Signed-off-by: Federico Gimenez <fgimenez@redhat.com>

### CODE REVIEW

**Comment by fgimenez** [comment]
This is ready for review, @dhiller pls take a look when you have some time :)

**Comment by fgimenez** [comment]
btw not sure why tide job has my pic in this PR! :D  Checked the rest and only happens here, maybe a GH hiccup

File: github/ci/prow-deploy/hack/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+set -e
+
+main(){
+    current_dir=$(dirname "$0")
+    project_infra_root=$(readlink -f "${current_dir}/../../../..")
+
+    base_dir=${project_infra_root}/github/ci/prow-deploy
+
+    source ${project_infra_root}/hack/manage-secrets.sh
+    decrypt_secrets
```
Comment (dhiller) [inline github/ci/prow-deploy/hack/deploy.sh:12]: nit: maybe we could add the target directory where to put the file and target file name as an argument (in another PR of course)?

**Review by dhiller — APPROVED** [review]
/approve
/hold

One suggestion inline for maybe a later PR :) Remove hold when feasible.

File: github/ci/prow-deploy/hack/deploy.sh (hunk 1 above)
Comment (fgimenez) [reply github/ci/prow-deploy/hack/deploy.sh:12]: Sure, sgtm :+1:

**Comment by fgimenez** [comment]
/unhold

thx!

**Pull request merged**